src/js/linked-heading.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/**
+ * Represents a filterable component heading.
+ */
+class LinkedHeading {
+
+	/**
+	 * Class constructor.
+	 * @param {HTMLElement} [headingElement] - The heading element in the DOM.
+	 */
+	constructor(headingElement) {
+		this.headingElement = headingElement;
+		this.id = headingElement.getAttribute('id');
+		this.buildLink();
+	}
+
+	/**
+	 * Build the heading link.
+	 */
+	buildLink() {
+		if (this.id) {
+			this.link = document.createElement('a');
+			this.link.setAttribute('href', `#${this.id}`);
+			this.link.setAttribute('title', 'Link directly to this section of the page');
+			this.link.innerHTML = '&nbsp;¶';
```

Comment:
Why is it a pilcrow?

Reply:
So that's what that's called!

Reply:
Because that's what we use in techdocs :)

Reply:
We could change it if we have a better candidate. Though after a quick look, [Wikipedia says](https://en.wikipedia.org/wiki/Pilcrow):
> Online, it is used in some blogs and wikis to denote permalinks[13] (cf. Purple Numbers).

Reply:
Interesting... I had a very different understanding of the pilcrow. Here's some fun reading :D http://www.shadycharacters.co.uk/2011/02/the-pilcrow-part-1/ (@JakeChampion)